habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -505,33 +496,58 @@ def get_all_scenedataset_receptacles(
     return receptacles
 
 
-def import_tri_mesh_ply(ply_file: str) -> TriangleMeshData:
+def import_tri_mesh(mesh_file: str) -> List[mn.trade.MeshData]:
     """
-    Returns a Tuple of (verts,indices) from a ply mesh using magnum trade importer.
+    Returns a list of MeshData objects from a mesh asset using magnum trade importer.
 
-    :param ply_file: The input PLY mesh file. NOTE: must contain only triangles.
+    :param mesh_file: The input meshes file. NOTE: must contain only triangles.
     """
     manager = mn.trade.ImporterManager()
     importer = manager.load_and_instantiate("AnySceneImporter")
-    importer.open_file(ply_file)
+    importer.open_file(mesh_file)
 
-    # TODO: We don't support mesh merging or multi-mesh parsing currently
-    if importer.mesh_count > 1:
-        raise NotImplementedError(
-            "Importing multi-mesh receptacles (mesh merging or multi-mesh parsing) is not supported."
-        )
+    mesh_data: List[mn.trade.MeshData] = []
 
-    mesh_ix = 0
-    mesh = importer.mesh(mesh_ix)
-    assert (
-        mesh.primitive == mn.MeshPrimitive.TRIANGLES
-    ), "Must be a triangle mesh."
+    if importer.scene_count > 0:
+        scene_id = importer.default_scene
+        # If there's no default scene, load the first one
+        if scene_id == -1:
+            scene_id = 0
 
-    # zero-copy reference to importer datastructures
-    mesh_data = TriangleMeshData(
-        mesh.attribute(mn.trade.MeshAttribute.POSITION),
-        mesh.indices,
-    )
+        scene = importer.scene(scene_id)
+
+        # Mesh referenced by mesh_assignments[i] has a corresponding transform in
+        # mesh_transformations[i]. Association to a particular node ID is stored in
+        # scene.mapping(mn.trade.SceneField.MESH)[i], but it's not needed for anything
+        # here.
+        mesh_assignments: cr.containers.StridedArrayView1D = scene.field(
+            mn.trade.SceneField.MESH
+        )
+        mesh_transformations: List[
+            mn.Matrix4
+        ] = mn.scenetools.flatten_transformation_hierarchy3d(
+            scene, mn.trade.SceneField.MESH
+        )
+        assert len(mesh_assignments) == len(mesh_transformations)
+
+        # A mesh can be referenced by multiple nodes, so this can't operate in-place.
+        flattened_meshes: List[mn.trade.MeshData] = [
+            mn.meshtools.transform3d(importer.mesh(mesh_id), transformation)
```

Comment:
If the same mesh is referenced more than once (such as is the case with chairs, etc.), this will unnecessarily perform the (complex) import multiple times. So probably better to do what you do in the `else` clause here as well, and only then update the meshes with the flattened transforms:

```py
    mesh_data: List[mn.trade.MeshData] = [
        importer.mesh(mesh_id) for mesh_id in range(importer.mesh_count)
    ]
    
    if importer.scene_count > 0:
        ...
    
        # A mesh can be referenced by multiple nodes, so this can't operate in-place
        # i.e., len(mesh_data) likely changes after this step
        mesh_data = [
            mn.meshtools.transform3d(mesh_data[mesh_id], transformation)
            for mesh_id, transformation in zip(
                mesh_assignments, mesh_transformations
            )
        ]
        
    # no else here
```